### PR TITLE
Reward creator from voting pool + proportional protocol rewards

### LINKED ISSUE #200: Reward 0.3% of curation voting pool to creator

### LINKED ISSUE #202: Distribute quadratic matched pool rewards in proportion to amount voted
- [x] add tests for edge cases, i.e: 1 curator, 1000 curators voting with 1 ATOM vs. 1 curator voting with 1000 ATOM

### PR DESCRIPTION
**Pull request by shanev**
Fixes #200. Fixes #202.

## Summary
* Add creator voting reward governance parameter
* Distribute 0.3% of voting pool to creator
* Distribute protocol reward in proportion to voting amount

### CODE REVIEW

File: x/curating/abci.go [hunk 1]
```diff
@@ -93,3 +98,22 @@ func EndBlocker(ctx sdk.Context, k keeper.Keeper) []abci.ValidatorUpdate {
 	}
 	return []abci.ValidatorUpdate{}
 }
+
+// sends matching reward in proportion to upvote
+func sendMatchingReward(ctx sdk.Context, k keeper.Keeper, upvoteAmount sdk.Int,
+	matchPoolPerVote sdk.Dec, rewardAccount sdk.AccAddress) error {
+
+	voteNum, err := upvoteAmount.QuoRaw(1_000_000).ToDec().ApproxSqrt()
```
Comment (jhernandezb) [inline x/curating/abci.go:106]: I think this assumes amount has a relation of `1:1`, but don't think this considers the quadratic votes

Comment (shanev) [reply x/curating/abci.go:106]: Let's walk through a case: If upvote is 9 ATOM, voteNum should be 3.
```
// because we store the coin in micros, the upvote amount is...
upvoteAmount = 9_000_000
// so we need to divide by 1,000,000 to normalize
temp = 9
// now we take the square root to come up with the original voteNum
voteNum = 3
```
How did you get 9?

Comment (jhernandezb) [reply x/curating/abci.go:106]: doh yeah ignore me is saturday night lol sounds good, I just noticed that we didn't store voteNum in the upvote struct  and got confused. 

Not sure if it would be better to store the voteNum inside each upvote as well, since the default VoteAmount is a Param that could potentially change (gov change required tho)

Comment (shanev) [reply x/curating/abci.go:106]: Yeah good point. I was trying to be too efficient and not store it, but that was premature optimization. We should store it: https://github.com/public-awesome/stakebird/issues/245.

Comment (shanev) [reply x/curating/abci.go:106]: Also 🍺:)

File: x/curating/abci_test.go [hunk 2]
```diff
@@ -85,22 +88,26 @@ func TestEndBlockerExpiringPost(t *testing.T) {
...
+	require.Equal(t, "10300000", creatorBal.AmountOf("ustb").String(),
+		"10 (bal) + 0.3 (creator protocol reward)")
 
 	curator1Bal := app.BankKeeper.GetAllBalances(ctx, addrs[1])
-	require.Equal(t, "14000000", curator1Bal.AmountOf("uatom").String(),
-		"9 (bal) + 5 (voting reward)")
-	require.Equal(t, "12850000", curator1Bal.AmountOf("ustb").String(),
-		"9 (bal) + 1 (deposit) + 2.85 (match reward)")
+	require.Equal(t, "13985000", curator1Bal.AmountOf("uatom").String(),
+		"9 (bal) + ? (voting reward)")
```
Comment (JakeHartnell) [inline x/curating/abci_test.go:102]: Did you mean to have a `?` hear or was that supposed to be a placeholder?

File: x/curating/abci_test.go [hunk 3]
```diff
@@ -85,22 +88,26 @@ func TestEndBlockerExpiringPost(t *testing.T) {
...
+	require.Equal(t, "11425000", curator1Bal.AmountOf("ustb").String(),
+		"9 (bal) + 1 (deposit) + 1.425 (match reward)")
 
 	curator2Bal := app.BankKeeper.GetAllBalances(ctx, addrs[2])
-	require.Equal(t, "6000000", curator2Bal.AmountOf("uatom").String(),
-		"1 (bal) + 5 (voter reward)")
-	require.Equal(t, "12850000", curator1Bal.AmountOf("ustb").String(),
-		"9 (bal) + 1 (deposit) + 2.85 (match reward)")
+	require.Equal(t, "5985000", curator2Bal.AmountOf("uatom").String(),
+		"1 (bal) + ? (voter reward)")
```
Comment (JakeHartnell) [inline x/curating/abci_test.go:108]: Same here. Might be nice to specify what the voter reward should be.

File: x/curating/abci_test.go [hunk 4]
```diff
@@ -124,16 +131,16 @@ func TestEndBlockerExpiringPostWithSmolRewardPool(t *testing.T) {
 		"10 (initial) + 0.3 (creator match reward)")
 
 	curator1Bal := app.BankKeeper.GetAllBalances(ctx, addrs[1])
-	require.Equal(t, "14000000", curator1Bal.AmountOf("uatom").String(),
-		"9 (bal) + 5 (voting reward)")
-	require.Equal(t, "10000474", curator1Bal.AmountOf("ustb").String(),
-		"9 (bal) + 1 (deposit) + 474u (match reward)")
+	require.Equal(t, "13985000", curator1Bal.AmountOf("uatom").String(),
+		"9 (bal) + ? (voting reward)")
```
Comment (JakeHartnell) [inline x/curating/abci_test.go:135]: Ditto.

**Review by JakeHartnell — APPROVED** [review]
LGTM. See comments.

File: x/curating/abci_test.go (hunk 2 above)
Comment (shanev) [reply x/curating/abci_test.go:102]: Ah good catch. Will fill in.
